projects/swimlane/ngx-charts/src/lib/common/timeline/timeline.component.ts

```diff
@@ -107,14 +109,14 @@ export class Timeline implements OnChanges {
     }
 
     let domain = [];
-    if (this.scaleType === ScaleType.Time) {
-      const min = Math.min(...values);
-      const max = Math.max(...values);
+    if (this.scaleType === 'time') {
+      const min = this.xScaleMin ?? Math.min(...values);
+      const max = this.xScaleMax ?? Math.max(...values);
```

Comment:
This will only work partially, because e.g. 0 will be filtered but an empty string will not. I am attempting to fix this problem in the PR #1627 for the min max values of both axis. 
Bellow my approach for limiting X axis of a line chart, but can't tell if this works in the timeline component as well. 
```
const min = (isNaN(this.xScaleMin) ? undefined : this.xScaleMin) ?? Math.min(...values);
const max = (isNaN(this.xScaleMax) ? undefined : this.xScaleMax) ?? Math.max(...values);
```